ci/infra/testrunner/skuba/skuba.py

```diff
@@ -104,12 +105,30 @@ def join_nodes(self, masters=None, workers=None):
         if workers is None:
             workers = self.platform.get_num_nodes("worker")
 
-        for n in range(1, masters):
-            self.node_join("master", n)
-
-        for n in range(0, workers):
-            self.node_join("worker", n)
-
+        nodes = [("master", n) for n in range(1, masters)] + \
+                [("worker", n) for n in range(0, workers)] 
+        for role, node in nodes:
+            self.node_join(role, node)
+            self._wait_node_ready(role, node, timeout=180, backoff=20)
+
+
+    def _wait_node_ready(self, role, node, timeout=60, backoff=10):
```

Comment:
i'd call it `_wait_node_joined` as ready implies the `Ready` status which we don't check here